surch/surch.py

```diff
@@ -1,165 +1,330 @@
+        self.find_problematic_commits(directories_list, strings_list_to_search, self.log_path)
+        lgr.info('Search time: ' + print_performance(start, time.time()) + ' seconds')
+
+    def get_directory_list(self):
+        """ Get list of the clone directory in the path"""
+        full_path_list = []
+        for item in os.listdir(self.local_path):
+            full_path_list.append(os.path.join(self.local_path, item))
+        return full_path_list
+
+    def _create_search_strings(self):
+        """ Create part of the grep command from search list"""
+        search_strings = ''
+        for string in self.search_list:
             search_strings = "{0} --or -e '{1}'".format(search_strings, string)
         return search_strings
 
-    def get_directory_list(self, path=os.getcwd(), skipped_list=None):
-        ''' get list of the clone directory'''
-        skipped_list = skipped_list or []
-        repo_list = []
-        full_path_list = []
-        for item in os.listdir(path):
-            full_path_list.append(os.path.join(path, item))
-        all_repo = [d for d in full_path_list if os.path.isdir(d)]
-        for repo in all_repo:
-            if repo not in skipped_list:
-                repo_list.append(repo)
-        return repo_list
-
-    def search_in_commits(self, ignore_repo, local_path, search_list):
-        ''' save the blob url of problematic commits '''
-        start = time.time()
-        skipped_repo = self.get_skipped_repo(ignore_repo, local_path)
-        directories = self.get_directory_list(local_path, skipped_list=skipped_repo)
-        string_to_search = self._create_search_strings(search_list)
-        with open('bad_comm', 'w') as db:
-            db.writelines(self.find_problematic_commits(directories, string_to_search))
-        print 'search_in_commit time: ' + print_performance(start, time.time()) + ' seconds'
-
-    def find_problematic_commits(self, directories, string_to_search):
-        ''' search secret string in the commits file and save the blob_url of the bad files '''
-        urls_blob = []
+    def find_problematic_commits(self, directories, string_to_search, output_file_path):
+        """ Search secret string in the commits file and save the blob_url of the bad files """
         for directory in directories:
-            print directory
+            lgr.info('Now scan the {0} directory'.format(directory))
             repo_name = directory.split('/', -1)[-1]
-            urls_blob += self.find_problematic_commits_in_directory(directory, string_to_search, repo_name,
-                                                                    self.organization)
-        return urls_blob
+            commits_urls = self.find_problematic_commits_in_directory(directory, string_to_search, repo_name)
+            for commit_url in commits_urls:
+                with open(output_file_path, 'a') as db:
+                    db.write(commit_url + '\n')
```

Comment:
replace with tinydb.
output should be a json file containing a list containing a repo name and blob url for each find.